src/mozilla_linux_pkg_manager/cli.py

```diff
@@ -1,148 +1,73 @@
-        args (list, optional): the args to pass to ``func``.  Defaults to ()
-        kwargs (dict, optional): the kwargs to pass to ``func``.  Defaults to
-            {}.
-        sleeptime_kwargs (dict, optional): the kwargs to pass to ``sleeptime_callback``.
-            If None, use {}.  Defaults to None.
-    Returns:
-        object: the value from a successful ``function`` call
-    Raises:
-        Exception: the exception from a failed ``function`` call, either outside
-            of the retry_exceptions, or one of those if we pass the max
-            ``attempts``.
-    """
-    kwargs = kwargs or {}
-    attempt = 1
-    while True:
-        try:
-            return await func(*args, **kwargs)
-        except retry_exceptions:
-            attempt += 1
-            check_number_of_attempts(attempt, attempts, func, "retry_async")
-            await asyncio.sleep(
-                define_sleep_time(
-                    sleeptime_kwargs, sleeptime_callback, attempt, func, "retry_async"
-                )
-            )
+# Some retriable errors don't have their own custom exception in api_core.
+ADDITIONAL_RETRYABLE_STATUS_CODES = (408,)
 
 
-def check_number_of_attempts(
-    attempt: int, attempts: int, func: Callable[..., Any], retry_function_name: str
-) -> None:
-    if attempt > attempts:
-        logging.warning(f"{retry_function_name}: {func.__name__}: too many retries!")
-        raise
-
-
-def define_sleep_time(
-    sleeptime_kwargs: Optional[dict[str, Any]],
-    sleeptime_callback: Callable[..., int],
-    attempt: int,
-    func: Callable[..., Any],
-    retry_function_name: str,
-) -> float:
-    sleeptime_kwargs = sleeptime_kwargs or {}
-    sleep_time = sleeptime_callback(attempt, **sleeptime_kwargs)
-    logging.debug(
-        "{}: {}: sleeping {} seconds before retry".format(
-            retry_function_name, func.__name__, sleep_time
-        )
-    )
-    return sleep_time
+def should_retry(exc):
+    """Predicate for determining when to retry."""
+    if isinstance(exc, RETRYABLE_TYPES):
+        return True
+    elif isinstance(exc, api_exceptions.GoogleAPICallError):
+        return exc.code in ADDITIONAL_RETRYABLE_STATUS_CODES
+    elif isinstance(exc, auth_exceptions.TransportError):
+        return should_retry(exc.args[0])
+    else:
+        return False
+
 
+ASYNC_RETRY = retry_async.AsyncRetry(predicate=should_retry)
 
-async def batch_delete_versions(versions, args):
+
+async def batch_delete_versions(targets, args):
     client = artifactregistry_v1.ArtifactRegistryAsyncClient()
-    for package in versions:
-        batches = batched(versions[package], 50)
+    for package in targets:
+        batches = batched(targets[package], 50)
         for batch in batches:
             logging.info(
-                f"Deleting {format(len(batch), ',')} expired package versions for {package}."
+                f"Deleting {format(len(batch), ',')} expired package versions of {os.path.basename(package)}"
```

Comment:
Please change this log message when in dry run mode - "Deleting ..." is not accurate for that case. "Would delete ..." or something similar would be better.